packages/grid/x-data-grid-pro/src/hooks/features/lazyLoader/useGridLazyLoader.ts

```diff
@@ -0,0 +1,220 @@
+import * as React from 'react';
+import {
+  useGridApiEventHandler,
+  GridFeatureModeConstant,
+  GridRenderedRowsIntervalChangeParams,
+  useGridSelector,
+  gridSortModelSelector,
+  gridFilterModelSelector,
+  useGridApiOptionHandler,
+  GridEventListener,
+  GridRowEntry,
+  GridDimensions,
+  GridFeatureMode,
+} from '@mui/x-data-grid';
+import { useGridVisibleRows } from '@mui/x-data-grid/internals';
+import { getRenderableIndexes } from '@mui/x-data-grid/hooks/features/virtualization/useGridVirtualScroller';
+import { GridApiPro } from '../../../models/gridApiPro';
+import {
+  DataGridProProcessedProps,
+  GridExperimentalProFeatures,
+} from '../../../models/dataGridProProps';
+import { GridFetchRowsParams } from '../../../models/gridFetchRowsParams';
+
+function findSkeletonRowsSection(
+  visibleRows: GridRowEntry[],
+  range: { firstRowIndex: number; lastRowIndex: number },
+) {
+  let { firstRowIndex, lastRowIndex } = range;
+  let isSkeletonSectionFound = false;
+
+  while (!isSkeletonSectionFound && firstRowIndex < lastRowIndex) {
+    if (!visibleRows[firstRowIndex].model && !visibleRows[lastRowIndex].model) {
```

Comment:
The first demo in https://deploy-preview-5214--material-ui-x.netlify.app/x/react-data-grid/row-updates/#lazy-loading is crashing here when I scroll all the way to the bottom.

Reply:
I knew there was a reason why I used an additional set of indexes to track this loop. The tests were failing last night. I reverted the change and now it is working as expected.

CC @cherniavskii